src/Simulation/Simulators.Tests/QubitManagerTests.cs

```diff
@@ -122,18 +122,50 @@ public void TestQubitManager()
                 n_q = qm.Borrow(0, exclusion);
                 Assert.True(n_q.Length == 0);
             }
+        }
 
-            // Test allocating qubits over capacity
-            OperationsTestHelper.IgnoreDebugAssert(() =>
-            {
-                IQArray<Qubit> n_q;
+        /// <summary>
+        /// Tests for exceptional cases or over allocation.
+        /// </summary>
+        [Fact]
+        public void TestAllocateTooManyQubits()
+        {
```

Comment:
It is Ok, but I would not take this (and below) out as a complete separate test. I would keep it as a part of the above test function, just make it a separate scope {} with its own Qubit manager. Reason: not to add 4 new tests to the "set" of tests in the TestExplorer window. Kind of polutes it. If it grows by 4 with every little test, the list becomes too long. Up to you, though.

Reply:
I've always tended toward more tests with smaller scopes since it allows targeted investigation by running a single test case either from the editor or the cmdline, though I can see the argument the other way. I'll move these back into the above function.